### PR TITLE
Bleed page improvements, valves now reflect the correct switch position + placeholder logic for packs

### PR DESCRIPTION
**Pull request by nikotina91**
**Summary of Changes**
ECAM bleed page changes:
- xbleed valve indicator now matches the overhead switch position
- left pack valve indicator now matches overhead switch position
- right pack valve indicator now matches overhead switch position
- ram air valve indicator now matches overhead switch position
- added AUTO position logic for xbleed switch, asked Komp on discord about this.
- Added LO/HI pack indicators, made them reflect overhead switch position (need feedback on this especially as I'm not sure of the functionality)
- Added C/H pack indicators, static, need feedback on functionality
- Added placeholder logic for engine bleed temperatures and pressures
- Added placeholder logic for pack inflow and outflow temperatures
- centered "ram", it was offset to the left, not sure if intended but i've see pics where it is centered
- moved "c" text, it was not aligned with the rest

**Screenshots (if necessary)**
before:
before

xbleed and ram air valves closed with both engines running with bleeds open:
Screenshot 2020-10-04 145004

All valves closed:
Screenshot 2020-10-04 145037

Engine 2 on with xbleed open:
Screenshot 2020-10-04 145103

Apu bleed on and both engine bleeds off:
Screenshot 2020-10-04 145934

**References**
- website about a320 ecam: http://www.a320dp.com/A320_DP/pneumatics/sys-15.1.3.html
ref1
- discord:
Screenshot 2020-10-04 151548
-A320 sim pilot video about the mod:
https://youtu.be/Zsk72luNZvc?t=397

**Additional context**
The logic for the temperatures and pressures is supposed to be a placeholder while ThePilot finishes the cond-packs system, can be easily remove if not fitting/not wanted

Discord username (if different from GitHub):

### CODE REVIEW

**Comment by camilo625** [comment]
Name: Camilo625 #5575
Date of testing: Current date 04/10/2020
Version of the sim: 1.9.3.0
PR Tested: #1083 

Testing technique: Spawn and check bleed page
Steps you've made to test the Pull Request:
1. Spawn C&D
2. Check bleed with APU
3. Check bleed with engines
4. Check bleed with PACKs

Overall Rating: Good
Conclusions of the testing: Works as described, great work.

**Comment by Benjozork** [comment]
Hello,

> website about a320 ecam: http://www.a320dp.com/A320_DP/pneumatics/sys-15.1.3.html

A320DP is not a very reliable source, as it mostly applies to the EIS1 display system. In this case, I found [this](https://www.youtube.com/watch?v=skR1DCASjPQ) for the EIS2 system, so you're good as it looks almost identical.

Just for future reference, avoid this site :)

**Comment by Benjozork** [comment]
(nitpick) this is not aligned properly:

image

irl: 
image

**Comment by nikotina91** [comment]
> Hello,
> 
> > website about a320 ecam: http://www.a320dp.com/A320_DP/pneumatics/sys-15.1.3.html
> 
> A320DP is not a very reliable source, as it mostly applies to the EIS1 display system. In this case, I found [this](https://www.youtube.com/watch?v=skR1DCASjPQ) for the EIS2 system, so you're good as it looks almost identical.
> 
> Just for future reference, avoid this site :)

alright, i'll keep that in mind, thanks



> (nitpick) this is not aligned properly:
> 
> image
> 
> irl:
> image

no worries, will fix after dinner

**Comment by lukecologne** [comment]
Also looks like the pack indicators look a bit off: 
grafik
IRL:
grafik
The arcs should be flatter, and maybe round the edges of the indicator lines a bit

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js [hunk 1]
```diff
@@ -79,6 +84,135 @@ var A320_Neo_LowerECAM_BLEED;
                 this.apuBleedIndication[0].setAttribute("visibility", "hidden")
                 this.apuBleedIndication[1].setAttribute("visibility", "visible")
             }
+
+            //find left pack status
+            var currentLeftPackState = SimVar.GetSimVarValue("L:A32NX_AIRCOND_PACK1_TOGGLE", "bool")
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js:89]: Do **not** use vars, if you are getting a SimVar, or setting a variable that won't change use `const`; if you are setting a variable that will change use `let`.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js [hunk 2]
```diff
@@ -79,6 +84,135 @@ var A320_Neo_LowerECAM_BLEED;
...
+            if(currentLeftPackState){
+                this.leftPack[0].setAttribute("visibility", "visible")
+                this.leftPack[1].setAttribute("visibility", "hidden")
+            } else {
+                this.leftPack[0].setAttribute("visibility", "hidden")
+                this.leftPack[1].setAttribute("visibility", "visible")
+            }
+
+            //find right pack status
+            var currentRightPackState = SimVar.GetSimVarValue("L:A32NX_AIRCOND_PACK2_TOGGLE", "bool")
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js:100]: Use const here

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js [hunk 3]
```diff
@@ -79,6 +84,135 @@ var A320_Neo_LowerECAM_BLEED;
...
+            if(currentRightPackState){
+                this.rightPack[0].setAttribute("visibility", "visible")
+                this.rightPack[1].setAttribute("visibility", "hidden")
+            } else {
+                this.rightPack[0].setAttribute("visibility", "hidden")
+                this.rightPack[1].setAttribute("visibility", "visible")
+            }
+
+            //find xbleed switch state
+            var currentXbleedState = SimVar.GetSimVarValue("L:A32NX_KNOB_OVHD_AIRCOND_XBLEED_Position", "Position(0-2)")
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js:111]: Use const

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js [hunk 4]
```diff
@@ -79,6 +84,135 @@ var A320_Neo_LowerECAM_BLEED;
...
+            } else if(currentApuBleedSate && currentXbleedState == 1){
+                this.xBleed[0].setAttribute("visibility", "visible")
+                this.xBleed[1].setAttribute("visibility", "hidden")
+            } else {
+                this.xBleed[0].setAttribute("visibility", "hidden")
+                this.xBleed[1].setAttribute("visibility", "visible")
+            }
+
+            //find ram air state
+            var currentRamState = SimVar.GetSimVarValue("L:A32NX_AIRCOND_RAMAIR_TOGGLE", "bool")
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js:125]: Use const

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js [hunk 5]
```diff
@@ -67,8 +74,6 @@ var A320_Neo_LowerECAM_BLEED;
                 this.rightEngineIp[1].setAttribute("visibility", "visible")
             }
 
-
-
             //find if the APU bleed is on
             var currentApuBleedSate = SimVar.GetSimVarValue("BLEED AIR APU", "Bool")
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js:78]: Use const

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js [hunk 6]
```diff
@@ -79,6 +84,135 @@ var A320_Neo_LowerECAM_BLEED;
...
+            //mock logic for the bleed page temperatures and pressures, to be replaced/updated/removed when the cond-packs system is implemented
+
+            var eng1Running = SimVar.GetSimVarValue("ENG COMBUSTION:1", "bool")
+            var eng2Running = SimVar.GetSimVarValue("ENG COMBUSTION:2", "bool")
+
+            var eng1TMP = SimVar.GetSimVarValue("ENG EXHAUST GAS TEMPERATURE:1", "Rankine")      
+            var eng1PSI = SimVar.GetSimVarValue("TURB ENG BLEED AIR:1", "Ratio (0-16384)")
+
+            var eng2TMP = SimVar.GetSimVarValue("ENG EXHAUST GAS TEMPERATURE:2", "Rankine")
+            var eng2PSI = SimVar.GetSimVarValue("TURB ENG BLEED AIR:2", "Ratio (0-16384)")
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js:161]: Use const here

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js [hunk 7]
```diff
@@ -79,6 +84,135 @@ var A320_Neo_LowerECAM_BLEED;
...
+
+            var xBleedValveOpen = false
+
+            if((currentXbleedState == 2 || (currentXbleedState == 1 && currentApuBleedSate))){
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js:165]: Replace with `if(currentXbleedState == 2 || (currentXbleedState == 1 && currentApuBleedSate)){`

You had an extra pair of brackets which werent needed

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js [hunk 8]
```diff
@@ -79,6 +84,135 @@ var A320_Neo_LowerECAM_BLEED;
...
+            var eng1Running = SimVar.GetSimVarValue("ENG COMBUSTION:1", "bool")
+            var eng2Running = SimVar.GetSimVarValue("ENG COMBUSTION:2", "bool")
+
+            var eng1TMP = SimVar.GetSimVarValue("ENG EXHAUST GAS TEMPERATURE:1", "Rankine")      
+            var eng1PSI = SimVar.GetSimVarValue("TURB ENG BLEED AIR:1", "Ratio (0-16384)")
+
+            var eng2TMP = SimVar.GetSimVarValue("ENG EXHAUST GAS TEMPERATURE:2", "Rankine")
+            var eng2PSI = SimVar.GetSimVarValue("TURB ENG BLEED AIR:2", "Ratio (0-16384)")
+
+            var xBleedValveOpen = false
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js:163]: Use let here

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js [hunk 9]
```diff
@@ -79,6 +84,135 @@ var A320_Neo_LowerECAM_BLEED;
...
+            if(currentRamState) {
+                this.ramAir[0].setAttribute("visibility", "hidden")
+                this.ramAir[1].setAttribute("visibility", "visible")
+            } else {
+                this.ramAir[0].setAttribute("visibility", "visible")
+                this.ramAir[1].setAttribute("visibility", "hidden")
+            }
+
+             //find pack flow state
+             var currentPackFlow = SimVar.GetSimVarValue("L:A32NX_KNOB_OVHD_AIRCOND_PACKFLOW_Position", "Position(0-2)")
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js:136]: Use const

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js [hunk 10]
```diff
@@ -79,6 +84,135 @@ var A320_Neo_LowerECAM_BLEED;
...
+            if((currentXbleedState == 2 || (currentXbleedState == 1 && currentApuBleedSate))){
+                xBleedValveOpen = true
+            }else{
+                xBleedValveOpen = false
+            }
+
+            if(currentEngineBleedState[0] && eng1Running){
+                this.querySelector("#eng1-bleed-tmp").textContent = parseInt(((eng1TMP - 491.67) * (5 / 9)) / 3.5)
+                this.querySelector("#eng1-bleed-psi").textContent = parseInt(eng1PSI)
+                } else {
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js:174]: Remove extra indent here, has issues with code readability

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js [hunk 11]
```diff
@@ -79,6 +84,135 @@ var A320_Neo_LowerECAM_BLEED;
...
+                this.querySelector("#eng1-bleed-tmp").textContent = "XXX"
+                this.querySelector("#eng1-bleed-psi").textContent = "xx"
+            }
+
+            if(currentLeftPackState && currentEngineBleedState[0] && eng1Running){
+                this.querySelector("#left-pack-in").textContent = parseInt(((eng1TMP - 491.67) * (5 / 9)) / 9)
+                this.querySelector("#left-pack-out").textContent = parseInt(((eng1TMP - 491.67) * (5 / 9)) / 20)
+                console.log("catch")
```
Comment (Meqolo) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js:182]: Recommend removing this console.log for the push

**Review by Meqolo — CHANGES_REQUESTED** [review]
Please add spaces between parts of ifs, ie you have places where you use `}else{`, instead please use `} else {`, or `if() {` should be `if () {`

**Comment by nikotina91** [comment]
Will do, at the moment I'm changing a few things  but I'll definitely fix all of those

**Comment by nikotina91** [comment]
- changed var to let
- centered "GND", "HI/LO", "RAM AIR"
- fixed spacing (hopefully didnt miss anything)
- packs should now show outflow temperatures based on the cond knob with the lowest setting (between 18° and 30°)
Screenshot 2020-10-04 211955
ref2
- now setting the L:x_bleed_valve simvar based on the bleed switch position

will do: flatter pack indicator arcs and rounded edges

**Comment by nikotina91** [comment]
afraid I'm not able to flatten the arcs as I have absolutely no idea on how to manipulate svg paths

**Comment by Benjozork** [comment]
> afraid I'm not able to flatten the arcs as I have absolutely no idea how to manipulate svg paths

Would you like to work with me on this ? I have worked with svg paths recently.

**Comment by nikotina91** [comment]
> > afraid I'm not able to flatten the arcs as I have absolutely no idea how to manipulate svg paths
> 
> Would you like to work with me on this ? I have worked with svg paths recently.

I've figured it out thanks! was getting very confused by whatever this was: 
ref4

while this did the trick:
ref6

this is how it looks now:
ref5

**Comment by nikotina91** [comment]
this is how it looks now:
ref6

based on: https://www.youtube.com/watch?v=skR1DCASjPQ

**Comment by sphectrum** [comment]
> 
> 
> this is how it looks now:
> ref6
> 
> based on: https://www.youtube.com/watch?v=skR1DCASjPQ

Could you please shorten the lower indicator needle just a little bit to match the upper one? The upper looks like a match with the IRL photo. Looks great btw!

**Comment by nikotina91** [comment]
few small changes based on previous comment:

ref7
ref8

**Comment by nikotina91** [comment]
yeah no clue what's going on here, need help

**Comment by lousybyte** [comment]
https://github.com/flybywiresim/a32nx/blob/2b5a177a7f81a9d842114223dcea2ebec2865412/A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.js#L135

Missing curly brace.

**Comment by nikotina91** [comment]
Final changelog due to mess above:

**Graphic changes:**
- xbleed valve indicator now matches the overhead switch position
- left pack valve indicator now matches overhead switch position
- right pack valve indicator now matches overhead switch position
- ram air valve indicator now matches overhead switch position
- added AUTO position logic for xbleed switch.
- Added LO/HI pack indicators, made them reflect overhead switch position
- Added C/H pack indicators, made them reflect the position of the switch set to the lowest
- aligned "RAM AIR"
- aligned "GND"
- moved "c" text, it was not aligned with the rest
- flattened indicator arcs to be more in line with reference and added small center dot
- increased font size for pack temperatures to 24

**placeholder logic for temperatures and pressures**
- Added placeholder logic for engine bleed temperatures and pressures
- engine bleed temperature based on "ENG EXHAUST GAS TEMPERATURE" simvar
- engine bleed temperature based on "TURB ENG BLEED AIR" simvar
- L:x_bleed_valve is set accordingly to the switch position
- Added placeholder logic for pack inflow and outflow temperatures
- inflow temperature based on engine bleed temperature
- outflow temperature of the pack will be set to the lowest setting of the 3 knobs, between 18° and 30°
- outflow temperature variation speed now influenced by pack flow switch (80% to 120%)
- engine bleeds now stay close regardless of the overhead switch status if the corresponding engine is off

**TO-DO in another PR if this ever gets through (don't want to add to much in a single one):**
- temperatures change to amber if above or below certain thresholds
- pressures change to amber if above or below certain thresholds
- valves and indicators change to amber if not in accord with real position (probably will have to wait for the real system to be 
   implemented)
- have certain valves close or open based on specific conditions regardless of switch position
- pack flow set to HIGH or LOW based on specific conditions regardless of switch position

**Comment by Benjozork** [comment]
I think this is good for testing,

**Comment by lawfulrocky** [comment]
Name: unlawfulrocky#4343
Date of testing: Current date DD/MM/YYYY
Version of the sim: Lastest dev
PR Tested: #1083

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1. Did APU bleed	
2. Turn on and of all bleeds	
3. Move Cockpit,fwd ,and Aft cabin knobs to hot and cold
4. Move Pack Flow to low to high 
5. See if temps and psi go up

Overall Rating: Great
Conclusions of the testing: See it works as intended. The ecam page is very interactive and show temps and psi corrective

**Review by Carter633 — COMMENTED** [review]
Name: Carter633 #1436
Date of testing: Current date 07/10/2020
Version of the sim: 1.9.3.0
PR Tested: #1083

Testing technique: Spawn Cold&Dark at EGKK, Start up one step at a time
Steps you've made to test the Pull Request:
1. Spawned Cold&Dark - external power with GPU initially, waited for all test pages to complete
2. Entered Bleed Page - verified all off, reading correctly
3. Started APU - watched APU Bleed change to open when I turned APU bleed on when avail. Watched pack temps increase and settle
4. Started ENG2 watched temps and pressures rise and settle
5. Started ENG1 watched Ts&Ps rise and settle.
6. Changed the cabin/cockpit temp setting dials in various ways - checked the needle holding lowest selected temp etc
7. Noted when ENG2 start was complete, Pack Outflow temps all read NaN for the remainder of the test
8. Checked behaviour when turning different packs off, and changing fan speed etc. All work as described.

Overall Rating: Needs Work - see below and point 7
Conclusions of the testing: Really good work on the bleed page. NaN reading erroneous I believe (but may be wrong - though sure it should continue to read an outflow temp?). Everything else seems to work as expected though. See attached image

Microsoft Flight Simulator 07_10_2020 09_38_33 (2)

**Comment by nikotina91** [comment]
It wasn't intentional no, I (hopefully for real this time) fixed the NaN thing about 20 minutes ago, it was due to the plane starting with some variables set to 0 and then going insane during cold and dark startups.

**Comment by Carter633** [comment]
Ah nice - want me to retest right away?

**Comment by nikotina91** [comment]
if you want/have time

**Comment by Carter633** [comment]
Currently re-testing. Now the second I hit APU Bleed, the input pack temp is already at 100 C immediately, and I'm still getting NaN from the moment I start engine start onwards unfortunately

**Comment by nikotina91** [comment]
alright, I tested it 2/3 times from cold and dark and it was going fine, i'll mark this as draft and take a better look at it, about the apu input temp its set to 00 as there's currently no simvar for that available I believe, i'll try to make it smoother anyway. thanks for testing!

**Comment by FlyingCarpet7** [comment]
Hi! Hope you don't mind if I give e few inputs. You might already know about it and it might not be addressed in this PR. 
Nice work so far!

ENG number 1 and 2 is in white all the time. Should be amber when N2 below idle. 
ANTI ICE symbol missing when Wing anti Ice is on.
IP HP APU and other things should be in white, not blue.
The lines under IP, HP and APU should be amber if there is no pneumatic feed available 
The fonts, specially on the numbers, seems to be a bit off.
°C on right pack could go a bit more to the right.
I also got the NaN fault.
In flight the middle parts with APU/GND should not be displayed.
BLEEDNEO

**Comment by nikotina91** [comment]
I think I will address most of those in a future PR, for now I've changed the color of the IP/HP text and moved the °C

**Review by Carter633 — APPROVED** [review]
Name: Carter633 #1436
Date of testing: Current date 01/10/2020
Version of the sim: 1.9.3.0
PR Tested: #1083

Testing technique: Full flight from Cold and Dark
Steps you've made to test the Pull Request:
1. Spawned at EGPH - full flight, Ext Pwr - bleed page, APU on and then APU bleed on
2. Checked Ts & Ps
3. Engine Start - checked Ts & Ps - NaN no longer an issue - good readings
4. No issues found!

Overall Rating: Awesome!
Conclusions of the testing: No issues found!

**Comment by AdenFlorian** [comment]
LGTM! Thanks!

**Pull request merged**